lms/static/js/commerce/track_ecommerce_events.js

```diff
@@ -0,0 +1,40 @@
+/**
+ *
+ * A library of helper functions to track ecommerce related events. 
+ *
+ */
+
+(function(define) {
+    'use strict';
+
+    define([
+        'jquery'
+    ], function($) {
+        var trackUpsellClick = function(linkName, optionalAttrs) {
+            if (!window.analytics) {
+                return;
+            }
+    
+            var eventAttrs = { "linkName": linkName };
+
+            if (
+                typeof optionalAttrs !== 'undefined' &&
+                optionalAttrs !== null &&
+                Object.keys(optionalAttrs).length > 0
+            ) {
+                var allowedAttrs = ["courseId", "pageName"];
+                allowedAttrs.map(
+                    allowedAttr => eventAttrs[allowedAttr] = optionalAttrs[allowedAttr]
+                );
+            }
+    
+            window.analytics.track("edx.bi.ecommerce.upsell_links_clicked", eventAttrs);
+        };
+        
+        var TrackECommerceEvents = {
+            trackUpsellClick: trackUpsellClick,
+        };
+
+        return TrackECommerceEvents;
+    });
+}).call(this, define || RequireJS.define);
```

Comment:
Then for the links that share code, we we'll need to add logic into track_ecommerce_events.js - I imagine a little "_get_link_name" function into that takes page, category, and calling page, and gives back the link name. I put the various unique combos in the 'differentiation' tab here: https://docs.google.com/spreadsheets/d/1nIf032qylE01TrebGJYZ2E8Kz93sxkqTFksuf91wbL4/edit#gid=1968923299

Reply:
I was hoping to keep the naming logic in each page that adds this function to the button/link so we can keep this module generic.